File: app/src/main/java/com/example/phl/data/sensation/TactileSensationDao.java
package com.example.phl.data.sensation;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.Date;
import java.util.List;

@Dao
public interface TactileSensationDao {

    @Insert
    long insert(TactileSensation tactileSensation);

    @Query("SELECT * FROM tactile_sensation")
    List<TactileSensation> getAllTactileSensations();

    @Query("SELECT * FROM tactile_sensation WHERE id = :id")
    TactileSensation getTactileSensationById(int id);

    @Query("UPDATE tactile_sensation SET value = :value WHERE id = :id")
    void updateTactileSensation(int id, int value);

    @Query("DELETE FROM tactile_sensation WHERE id = :id")
    void deleteTactileSensation(int id);
}

File: app/src/main/java/com/example/phl/data/sensation/TactileSensationDao.kt
package com.example.phl.data.sensation

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.example.phl.data.AbstractDao
import java.time.LocalDate

@Dao
interface TactileSensationDao : AbstractDao<TactileSensation> {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    override fun insert(data: TactileSensation)

    @Query("SELECT * FROM tactile_sensation")
    override fun getAll(): List<TactileSensation>

    @Query("SELECT * FROM tactile_sensation where sessionId = :sessionId")
    override fun getBySessionId(sessionId: String): List<TactileSensation>

    @Query("SELECT * FROM tactile_sensation WHERE sessionId = :id")
    override fun getById(id: String): TactileSensation

    @Query("SELECT * FROM tactile_sensation WHERE strftime('%Y-%m-%d', time) = :day")
    override fun getByDay(day: LocalDate): List<TactileSensation>

    @Query("DELETE FROM tactile_sensation WHERE sessionId = :id")
    override fun deleteById(id: String)

    @Query("DELETE FROM tactile_sensation WHERE sessionId = :sessionId")
    override fun deleteBySessionId(sessionId: String)
}